pkg/openstack/loadbalancer.go

```diff
@@ -973,20 +1168,56 @@ func (lbaas *LbaasV2) getServiceAddress(clusterName string, service *corev1.Serv
 				PortID:            portID,
 				Description:       fmt.Sprintf("Floating IP for Kubernetes external service %s from cluster %s", serviceName, clusterName),
 			}
-			if svcConf.lbPublicSubnetID != "" {
-				floatIPOpts.SubnetID = svcConf.lbPublicSubnetID
-			}
-			if loadBalancerIP != "" {
-				floatIPOpts.FloatingIP = loadBalancerIP
-			}
 
-			klog.V(4).Infof("Creating floating ip with opts %+v", floatIPOpts)
+			if loadBalancerIP == "" && svcConf.lbPublicSubnetSpec.MatcherConfigured() {
+				var found_subnet subnets.Subnet
+				// tweak list options for tags
+				subnets, err := lbaas.listSubnetsForNetwork(svcConf.lbPublicNetworkID, svcConf.lbPublicSubnetSpec.TweakListOptsFunction())
```

Comment:
Variable 'subnets' collides with imported package name

Reply:
I'll replace this with `foundSubnets`